### PR TITLE
CONTRIBUTING.md: first draft of how to write and change spec.

### PR DESCRIPTION
**Pull request by rustyrussell**
Additions and feedback welcome!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+### Writing The Requirements
+
+Some requirements are obvious, some are subtle.  They're designed to walk
+an implementer through the code they have to write, so write them as YOU
+develop YOUR implementation.  Stick with `MUST`/`SHOULD`/`MAY` and `NOT`:
+see [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+Requirements are grouped into writer and reader, just as implementations
+are.  Make sure you define exactly what a writer must do, and exactly what
+a reader must do if it the writer doesn't do that!  A developer should
```
Comment (t-bast) [inline CONTRIBUTING.md]:
```suggestion
a reader must do if the writer doesn't do that!  A developer should
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+implementer?
+
+### Creating Test Vectors
+
+For new low-level protocol constructions, test vectors are necessary.
+These have traditionally been lines within the spec itself, but the modern
+trend is to use JSON and separate files.  The intent is that they be
+machine-readable by implementations.
+
+For new inter-peer messages, a test framework is in development to simulate
```
Comment (t-bast) [inline CONTRIBUTING.md]: Could you maybe link to its repository (if there is one)?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+Unknown odd inter-peer messages are ignored, aka "it's OK to be odd!"
+which makes more sense as you get to know me.
+
+If your message is an enhancement, and you don't need to know if the other
+side supports it, you should give it an odd number.  If it would be broken
+if the other side doesn't support it (ie. Should Never Happen) give it an
+even number.  Mistakes happen, and future versions of the software may well
+not be tested against ancient versions.
+
+If you want to experiment with new message types internally, I recommend
```
Comment (ariard) [inline CONTRIBUTING.md]: nit : maybe points to where message types are laid out like in [lightning message format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+using 32768 and above (use even, so it will break if these accidentally
+escape into the wild).
+
+### Adding New Feature Bits
+
+Feature bits are how you know a message is legal to send (see above), and
```
Comment (ariard) [inline CONTRIBUTING.md]: nit : maybe points to where features bits are laid out more in depth like in [init message](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#the-init-message)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+repository](https://github.com/lightningnetwork/lightning-rfc) for
+explicit issues and pull requests, and a bi-weekly IRC meeting on
+#lightning-dev on Freenode, currently held at 5:30am Tuesday, 
+Adelaide/Australia timezone (eg. Tuesday 23rd July 2019 05:30 == Mon, 22
+Jul 2019 20:00 UTC).
+
+Spelling, typo and formatting changes are accepted once two contributors
+ack and there are no nacks.  All other changes get approved and minuted at
+the IRC meeting.  Protocol changes require two independent implementations
+which successfully inter-operate.
```
Comment (ariard) [inline CONTRIBUTING.md]: and maybe add a sentence inviting people to be patient and polite if they PR to correct spec oddities and remembering them there are multiples implementations to coordinate and that takes time.

File: CONTRIBUTING.md (hunk 2 above)
Comment (rustyrussell) [reply CONTRIBUTING.md]: It'll be a PR RSN; it really needs to live in the main repository IMHO.

**Comment by cdecker** [comment]
Needs to be squashed, but otherwise an excellent addition :+1:

**Comment by rustyrussell** [comment]
Trivial rebase to fix conflict, fold fixes.

**Pull request merged**